### PR TITLE
nixos/mailcatcher: init module for existing package

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
I'm interested in running mailcatcher as a system service.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by aanderse** [comment]
@GrahamcOfBorg test mailcatcher

File: nixos/modules/services/mail/mailcatcher.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        User = "mailcatcher";
+        Restart = "always";
+        ExecStart = "${pkgs.mailcatcher}/bin/mailcatcher --foreground --no-quit --http-ip ${cfg.http.ip} --http-port ${toString cfg.http.port} --smtp-ip ${cfg.smtp.ip} --smtp-port ${toString cfg.smtp.port}";
+      };
+    };
+
+    users.users.mailcatcher = {
```
Comment (Ma27) [inline nixos/modules/services/mail/mailcatcher.nix]: If you don't need a home directory or have any reasons to create a dedicated user, you could use [`systemd's` dynamic user feature](http://0pointer.net/blog/dynamic-users-with-systemd.html).

Comment (aanderse) [reply nixos/modules/services/mail/mailcatcher.nix]: @Ma27 totally slipped my mind! Thank you for reminding me!

File: nixos/modules/services/mail/mailcatcher.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      };
+
+      smtp.ip = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        description = "The ip address of the smtp server.";
+      };
+
+      smtp.port = mkOption {
+        type = types.int;
```
[the remark this answers is not in the document]
Comment (aanderse) [reply nixos/modules/services/mail/mailcatcher.nix]: Was running on my local system (18.09) which doesn't have this yet, forgot to switch back. Thanks!

**Review by Ma27 — APPROVED** [review]

**Comment by Ma27** [comment]
@aanderse thanks!

**Comment by aanderse** [comment]
@Ma27 If you don't mind would you please also merge https://github.com/NixOS/nixpkgs/pull/56575? Thanks again!

**Pull request merged**